dbuild/__init__.py

```diff
@@ -0,0 +1,247 @@
+        host_config = docker_client.create_host_config(binds=binds)
+    else:
+        host_config = None
+
+    container = docker_client.create_container(
+        image=image, name=name, command=command, environment=env,
+        network_disabled=disable_network, volumes=volumes,
+        working_dir=cwd, host_config=host_config)
+    return container
+
+
+def start_container(docker_client, container):
+    """ Start docker container """
+    response = docker_client.start(container=container.get('Id'))
+    return response
+
+
+def wait_container(docker_client, container):
+    """ Wait  for the container to finish execution """
+    rv = docker_client.wait(container=container)
+    return rv
+
+
+def container_logs(docker_client, container):
+    """ Get container stdout and stderr """
+    return [log.strip() for log in docker_client.logs(container=container,
+                                                      stream=True,
+                                                      timestamps=True)]
+
+
+def remove_container(docker_client, container, force=False):
+    """ Remove docker container """
+    return docker_client.remove_container(container=container, force=force)
+
+
+def create_docker_dir(dist, release):
+    """Create docker directory and populate it"""
+    PATH = os.path.dirname(os.path.abspath(__file__))
+    TMPL_ENV = Environment(
+        autoescape=False,
+        loader=FileSystemLoader(os.path.join(PATH, 'templates')),
+        trim_blocks=False)
+
+    # Create docker_dir - a temporary directory which will have Dockerfile and
+    # scripts to build the container.
+    docker_dir = mkdtemp()
+    dockerfile = os.path.join(docker_dir, 'Dockerfile')
+
+    ctxt = {'dist': dist, 'release': release,
+            'maintainer': 'dbuild, dbuild@test.com'}
+
+    # Write Dockerfile under docker_dir
+    with open(dockerfile, 'w') as d:
+        dockerdata = TMPL_ENV.get_template('dockerfile.jinja').render(ctxt)
+        d.write(dockerdata)
+
+    # Copy scripts under docker_dir
+    shutil.copytree(os.path.join(PATH, 'scripts'),
+                    os.path.join(docker_dir, 'scripts'))
+    return docker_dir
+
+def docker_build(build_dir, build_type, source_dir='source', force_rm=False,
+                 docker_url='unix://var/run/docker.sock', dist='ubuntu',
+                 release='trusty', extra_repos_file='repos',
+                 extra_repo_keys_file='keys', build_cache=True):
+    c = docker_client(docker_url)
+
+    docker_path = create_docker_dir(dist, release)
+
+    print "Starting %s Package Build" % build_type
+    image_tag = 'dbuild-' + dist + '/' + release
+    response = build_image(c, docker_path, tag=image_tag, nocache=not build_cache)
+    print response
+    command=['bash', '-c', '']
+
+    if os.path.exists(os.path.join(build_dir, extra_repos_file)):
+        command[2] += 'cp /build/%s \
+        /etc/apt/sources.list.d/dbuild-extra-repos.list && ' % extra_repos_file
```

Comment:
I wanted third element to be conditionally populated (if extra_repos_file not exists, then the above one will not be set - only apt-get update/dist-upgrade will be added. So I dont see an easy way to create part of element conditionally.